cpp/src/cider/type/plan/CaseExpr.cpp

```diff
@@ -223,74 +223,86 @@ JITExprValue& CaseExpr::codegen(CodegenContext& context) {
       case_ti.is_fp() || case_ti.is_boolean()) {
     const auto type =
         case_ti.is_decimal() ? decimal_to_int_type(case_ti) : case_ti.get_type();
-    JITValuePointer value =
-        func.createVariable(getJITTag(type), "case_when_value_init", 0);
-    JITValuePointer null =
-        func.createVariable(JITTypeTag::BOOL, "case_when_null_init", false);
-    JITValuePointer is_case =
-        func.createVariable(JITTypeTag::BOOL, "is_case_init", false);
-    for (const auto& expr_pair : expr_pair_list) {
-      func.createIfBuilder()
-          ->condition([&]() {
-            cider::exec::nextgen::utils::FixSizeJITExprValue cond(
-                expr_pair.first->codegen(context));
-            auto condition = !cond.getNull() && cond.getValue();
-            return condition;
-          })
-          ->ifTrue([&]() {
-            cider::exec::nextgen::utils::FixSizeJITExprValue then_jit_expr_value(
-                expr_pair.second->codegen(context));
-            value.replace(then_jit_expr_value.getValue());
-            null.replace(then_jit_expr_value.getNull());
-            is_case = func.createLiteral(JITTypeTag::BOOL, true);
-          })
-          ->build();
-    }
+    JITValuePointer value = func.createVariable(getJITTag(type), "case_when_value_init");
+    *value = func.createLiteral(getJITTag(type), 0);
+    JITValuePointer null = func.createVariable(JITTypeTag::BOOL, "case_when_null_init");
```

Comment:
use this func to directly initialize？
https://github.com/intel/BDTK/blob/98e23618e4bc78629bf097003a9fd7b5cc2a6c3b/cpp/src/cider/exec/nextgen/jitlib/base/JITFunction.h#L92-L95

Reply:
It doesn't work, can't generate IR of if.